### ISSUE TITLE
JD-GUI still shows same contents for a recreated jar file

### ISSUE THREAD

**Issue by onacit** [java-decompiler/jd-gui#369]
When a jar file created and re-dragged (after the tab closed), the GUI shows same tree.

I need to shutdown the program and restart.
Do we have any caches or somethiin'?

**Comment by Flo354** [comment]
I noticed the same thing today.

**Comment by onacit** [comment]
I think I found it.

https://github.com/java-decompiler/jd-gui/blob/master/services/src/main/java/org/jd/gui/service/type/ClassFileTypeFactoryProvider.java#L50

**Comment by cwt9562** [comment]
i meet this bug too

**Comment by wurensen** [comment]
Any updating?

**Comment by HaleXie** [comment]
I met this issue too. I played with Proguard and I spent a few hours in finding out the cached classes in JD was shown instead of the latest version of the generate jars

**Comment by i2534** [comment]
yes, i met this too, with proguard, cost my many times to modify the rule, then nothing change...
then i search the mapping file, found the rule is right...
so i suspect jd use old cache...
copy the jar to a new location, then jd open right

**Comment by Boat2017** [comment]
现在是假期。已经收到您的来信，我会尽快回复。

**Comment by StevenJack666** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by zhonghongfly** [comment]
You can use jadx, it doesn't have this problem
